devito/data/data.py

```diff
@@ -188,15 +278,26 @@ def __setitem__(self, glb_idx, val):
                 # * one of them is 1
                 # Conceptually, below we apply the same rule
                 val_idx = val_idx[len(val_idx)-val.ndim:]
+                processed = []
+                # Handle step size > 1
+                for i, j in zip(glb_idx, val_idx):
+                    if isinstance(i, slice) and i.step is not None and i.step > 1 and \
```

Comment:
I've seen this pattern many times:

``` if isinstance(i, slice) and i.step is not None and i.step > ...```

maybe we should add a free function ``is_step_gt`` (ie, is step greater than) so that you can do

```
if is_step_gt(i, 1) and j.stop > j.start
```

or something along these lines.

Otherwise, four conditions in an ``if`` becomes a bit unreadable